lib/vision/index.js

```diff
@@ -0,0 +1,1378 @@
+ *   //   face: [...],
+ *   //   label: [...]
+ *   // }
+ * });
+ *
+ * //-
+ * // Supply multiple images for feature detection.
+ * //-
+ * var images = [
+ *   'image.jpg',
+ *   'image-two.jpg'
+ * ];
+ *
+ * var types = [
+ *   'face',
+ *   'label'
+ * ];
+ *
+ * vision.detect(images, types, function(err, detections, apiResponse) {
+ *   // detections = [
+ *   //   // Detections for image.jpg:
+ *   //   {
+ *   //     face: [...],
+ *   //     label: [...]
+ *   //   },
+ *   //
+ *   //   // Detections for image-two.jpg:
+ *   //   {
+ *   //     face: [...],
+ *   //     label: [...]
+ *   //   }
+ *   // ]
+ * });
+ */
+Vision.prototype.detect = function(images, options, callback) {
+  var self = this;
+  var isSingleImage = !is.array(images) || images.length === 1;
+
+  if (!is.object(options)) {
+    options = {
+      types: options
+    };
+  }
+
+  var types = arrify(options.types);
+
+  var typeShortNameToFullName = {
+    face: 'FACE_DETECTION',
+    faces: 'FACE_DETECTION',
+
+    label: 'LABEL_DETECTION',
+    labels: 'LABEL_DETECTION',
+
+    landmark: 'LANDMARK_DETECTION',
+    landmarks: 'LANDMARK_DETECTION',
+
+    logo: 'LOGO_DETECTION',
+    logos: 'LOGO_DETECTION',
+
+    properties: 'IMAGE_PROPERTIES',
+
+    safeSearch: 'SAFE_SEARCH_DETECTION',
+
+    text: 'TEXT_DETECTION'
+  };
+
+  Vision.findImages_(images, function(err, images) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    var config = [];
+
+    images.forEach(function(image) {
+      types.forEach(function(type) {
+        var typeName = typeShortNameToFullName[type];
+
+        if (!typeName) {
+          throw new Error('Requested detection feature not found: ' + type);
+        }
+
+        config.push({
+          image: image,
+          features: {
+            type: typeName
+          }
+        });
+      });
+    });
+
+    self.annotate(config, function(err, annotations, resp) {
+      if (err) {
+        callback(err, null, resp);
+        return;
+      }
+
+      function mergeArrayOfObjects(arr) {
+        return extend.apply(null, arr);
+      }
+
+      function formatAnnotationBuilder(type) {
+        return function(annotation) {
+          var formatMethodMap = {
+            faceAnnotations: Vision.formatFaceAnnotation_,
+            imagePropertiesAnnotation: Vision.formatImagePropertiesAnnotation_,
+            labelAnnotations: Vision.formatEntityAnnotation_,
+            landmarkAnnotations: Vision.formatEntityAnnotation_,
+            logoAnnotations: Vision.formatEntityAnnotation_,
+            safeSearchAnnotation: Vision.formatSafeSearchAnnotation_,
+            textAnnotations: Vision.formatEntityAnnotation_
+          };
+
+          return formatMethodMap[type](annotation, options);
+        };
+      }
+
+      var originalResp = extend(true, {}, resp);
+      var detections = [];
+
+      for (var i = 0; i < images.length; i++) {
+        detections.push(annotations.splice(0, types.length));
+      }
```

Comment:
This is more of a style nitpick.. so feel free to ignore this! I think we could cut down on a few lines if we ran `map` off of `images`.

```js
var detections = images.map(function() {
    return annotations.splice(0, types.length);
  })
  .map(mergeArrayOfObjects)
  .map(function(annotations) {});
```